src/components/rails/explorer-rail/pmBlocksLoop/pmBlocksLoop.vue

```diff
@@ -0,0 +1,156 @@
+<script>
+import nodeTypesStore from '@/nodeTypesStore';
+
+export default {
+  name: 'PmBlocksLoop',
+  data() {
+    return {
+      wasClicked: false,
+      element: null,
+      selectedItem: null,
+      xOffset: 0,
+      yOffset: 0,
+      movedElement: null,
+      helperStyles: {
+        backgroundColor:'#ffffff',
+        position: 'absolute',
+        height: '40px',
+        width: '40px',
+        zIndex: '10',
+        opacity: '0.5',
+        pointerEvents: 'none',
+      },
+    };
+  },
+  methods: {
+    onClickHandler(event, control) {
```

Comment:
Wasn't this the code you just refactored @TinCodes ?

Reply:
Yes it is, there is now a new Mixin (`src/mixins/clickAndDrop.js`) that should handle the drag and drop. The updated file `src/components/rails/explorer-rail/nodeTypesLoop/nodeTypesLoop.vue` should be able to provide an example of how it is being used.